docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -380,3 +390,99 @@ The Java code generated for this variant is:
     public Published(Long year, String publisher) { /* ... */ }
     public Record toValue() { /* ... */ }
   }
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
+A :ref:`parameterized type <daml-ref-parameterized-types>` is where the actual type is specified at instantiation as *type parameters* and not as part of type definition.
+The Java Code Generator uses Java Generic types to represent the DAML parameterized type.
+
+Below is a DAML fragment defining the parameterized type `Attribute` for use by the `
+
+.. code-block:: daml
```

Comment:
If possible, it's better to create a separate DAML file (in the same directory) and then pull the code in using a `literalinclude` block - that means we actually test whether the DAML compiles.